### PR TITLE
Nix refactor

### PR DESCRIPTION
**Pull request by nebunebu**
- Move default.nix into nix/default.nix to consolidate future potential nix files, such as those for a nixosModule for example.

- add esbuild to devShell. I have to add esbuild to the devshell and export AGS_BUNDLER=esbuild. (Really just done for personal convienence.)

- format nix files with nixfmt-rfc-style, since nixpkgs-fmt is being deprecated.

- simplify output function signature

### CODE REVIEW

**Comment by Jas-SinghFSU** [comment]
@mitsuruu Do you want to take this one?

File: flake.nix [hunk 1]
```diff
@@ -6,56 +6,75 @@
...
-        default = (pkgs.callPackage ./. { inherit inputs; }).desktop.script;
-      });
-  };
+      packages = forEachSystem (
+        system:
+        let
+          pkgs = pkgsFor.${system};
+        in
+        {
+          default = (pkgs.callPackage ./nix/default.nix { inherit inputs; }).desktop.script;
```
Comment (orxngc) [inline flake.nix:76]:
```suggestion
          default = (pkgs.callPackage ./nix { inherit inputs; }).desktop.script;
```

File: flake.nix [hunk 2]
```diff
@@ -6,56 +6,75 @@
...
-        pkgs = pkgsFor.${system};
-      in final: prev: {
-        hyprpanel = (pkgs.callPackage ./. { inherit inputs; }).desktop.script;
-      });
+      overlay = forEachSystem (
+        system:
+        let
+          pkgs = pkgsFor.${system};
+        in
+        final: prev: { hyprpanel = (pkgs.callPackage ./nix/default.nix { inherit inputs; }).desktop.script; }
```
Comment (orxngc) [inline flake.nix:67]:
```suggestion
        final: prev: { hyprpanel = (pkgs.callPackage ./nix { inherit inputs; }).desktop.script; }
```

File: flake.nix [hunk 3]
```diff
@@ -6,56 +6,75 @@
...
+              pkgsFor.${system}.dart-sass
+              pkgsFor.${system}.fd
+              pkgsFor.${system}.btop
+              pkgsFor.${system}.bluez
+              pkgsFor.${system}.bluez-tools
+              pkgsFor.${system}.grimblast
+              pkgsFor.${system}.gpu-screen-recorder
+              pkgsFor.${system}.brightnessctl
+              pkgsFor.${system}.gnome.gnome-bluetooth
+              pkgsFor.${system}.python3
```
Comment (orxngc) [inline flake.nix:46]:
```suggestion
              pkgsFor.${system}.python3
              pkgsFor.${system}.matugen
```

**Review by orxngc — COMMENTED** [review]
I think this all looks good. Not entirely sure though, I'm not that fluent in Nix.

**Comment by nebunebu** [comment]
@orxngc added your suggestions

File: flake.nix [hunk 4]
```diff
@@ -6,56 +6,76 @@
...
-      in final: prev: {
-        hyprpanel = (pkgs.callPackage ./. { inherit inputs; }).desktop.script;
-      });
+      overlay = forEachSystem (
+        system:
+        let
+          pkgs = pkgsFor.${system};
+        in
+        final: prev: { hyprpanel = (pkgs.callPackage ./nix { inherit inputs; }).desktop.script; }
+      );
```
Comment (isabelroses) [inline flake.nix:69]: Overlays should **never** need to be made per system
```suggestion
    overlay.default = final: prev: {
      hyprpanel = self.packages.${prev.stdenv.system}.default;
    });
```

**Review by isabelroses — COMMENTED** [review]
I can't leave a comment on it beacuse its not in the lines change.
```nix
export PATH=$PATH:${dart-sass}/bin
export PATH=$PATH:${fd}/bin
export PATH=$PATH:${btop}/bin
export PATH=$PATH:${pipewire}/bin
export PATH=$PATH:${bluez}/bin
export PATH=$PATH:${bluez-tools}/bin
export PATH=$PATH:${grimblast}/bin
export PATH=$PATH:${gpu-screen-recorder}/bin
export PATH=$PATH:${networkmanager}/bin
export PATH=$PATH:${brightnessctl}/bin
export PATH=$PATH:${pkgs.gnome.gnome-bluetooth}/bin
export PATH=$PATH:${python3}/bin
```
```nix
export PATH=$PATH:${lib.makeBinPath [dart-sass fd btop pipewire bluez bluez-tools grimblast gpu-screen-recorder brightnessctl gnome.gnome-bluetooth python3]}
```

**Comment by mitsuruu** [comment]
Apologies on taking so long to post a reply.

I think when the comments left by @isabelroses have been addressed it'll be good to merge in.

Additionally, thanks for the feedback @isabelroses, I'll definitely be taking both of the comments you left into account when refactoring some of my other flakes.

**Comment by nebunebu** [comment]
@isabelroses @mitsuruu

**Comment by Jas-SinghFSU** [comment]
@nebunebu When you get a chance could you please resolve merge conflicts from this branch? It'll prevent me from merging otherwise ;) thank you!

**Comment by isabelroses** [comment]
> I can't leave a comment on it beacuse its not in the lines change.
> 
> ```nix
> export PATH=$PATH:${dart-sass}/bin
> export PATH=$PATH:${fd}/bin
> export PATH=$PATH:${btop}/bin
> export PATH=$PATH:${pipewire}/bin
> export PATH=$PATH:${bluez}/bin
> export PATH=$PATH:${bluez-tools}/bin
> export PATH=$PATH:${grimblast}/bin
> export PATH=$PATH:${gpu-screen-recorder}/bin
> export PATH=$PATH:${networkmanager}/bin
> export PATH=$PATH:${brightnessctl}/bin
> export PATH=$PATH:${pkgs.gnome.gnome-bluetooth}/bin
> export PATH=$PATH:${python3}/bin
> ```
> 
> ```nix
> export PATH=$PATH:${lib.makeBinPath [dart-sass fd btop pipewire bluez bluez-tools grimblast gpu-screen-recorder brightnessctl gnome.gnome-bluetooth python3]}
> ```

sorry if I was unclear. I ment to prefer the final line here and I was intending for the change to go here https://github.com/Jas-SinghFSU/HyprPanel/blob/7ccb172bc76678be4a60eb3d20621fe214cc01a8/default.nix#L53 which should be changed, when using `mkShell` nix automatic does the path adjustment for you.

**Comment by nebunebu** [comment]
@isabelroses I hope this is better

**Review by isabelroses — APPROVED** [review]

**Comment by Jas-SinghFSU** [comment]
Thank you to everyone who contributed to this effort! You're awesome and are greatly appreciated :heart: . 

I'll go ahead and merge this shortly.

**Comment by Tragyn** [comment]
Has this made any significant changes to the installation process? Hyprpanel was working well previously, but I believe this merge may have changed how I'm meant to install it.  I'm new to NixOS so I can't say anything for certain, but here's the relevant section of my config:

`  nixpkgs.overlays = [
    inputs.hyprpanel.overlay.x86_64-linux
  ];`

pkgs.hyprpanel is no longer being found.

**Comment by mitsuruu** [comment]
> Has this made any significant changes to the installation process? Hyprpanel was working well previously, but I believe this merge may have changed how I'm meant to install it.  I'm new to NixOS so I can't say anything for certain, but here's the relevant section of my config:
> 
> 
> 
> `  nixpkgs.overlays = [
> 
>     inputs.hyprpanel.overlay.x86_64-linux
> 
>   ];`
> 
> 
> 
> pkgs.hyprpanel is no longer being found.
> 
> 
> 
> EDIT: Pinned to previous commit and it's working again.

Try removing the `.x86_64-linux` from the end of your overlay and try latest again? I believe one of the changes affected this but I will check if docs need updating tomorrow.

**Comment by Tragyn** [comment]
> > Has this made any significant changes to the installation process? Hyprpanel was working well previously, but I believe this merge may have changed how I'm meant to install it.  I'm new to NixOS so I can't say anything for certain, but here's the relevant section of my config:
> > `  nixpkgs.overlays = [
> > ```
> > inputs.hyprpanel.overlay.x86_64-linux
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ];`
> > pkgs.hyprpanel is no longer being found.
> > EDIT: Pinned to previous commit and it's working again.
> 
> Try removing the `.x86_64-linux` from the end of your overlay and try latest again? I believe one of the changes affected this but I will check if docs need updating tomorrow.

Thanks, that got me past those errors! hyprpanel still fails to build; I hope this is helpful:

`error: builder for '/nix/store/6dhfsi3qhl4qhc5fk4b4954mhsphdg8h-hyprpanel-latest.drv' failed with exit code 1;
       last 9 log lines:
       > Running phase: unpackPhase
       > unpacking source archive /nix/store/5x7bxfqa035biy5s4gfs5skxlfxjgnn5-nix
       > source root is nix
       > Running phase: patchPhase
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configurePhase
       > no configure script, doing nothing
       > Running phase: buildPhase
       > error: ModuleNotFound resolving "./main.ts" (entry point)
       For full logs, run 'nix log /nix/store/6dhfsi3qhl4qhc5fk4b4954mhsphdg8h-hyprpanel-latest.drv'.
error: 1 dependencies of derivation '/nix/store/9vxwq4kip2n9gqpm0zshg88xcvqnqx86-hyprpanel.drv' failed to build
error: 1 dependencies of derivation '/nix/store/9ip9c46adxk1l90290scmq99ci0yf02a-home-manager-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/nfhlxkaiwww8g11dsng0dly5x9g6g1zn-hyprpanel-fish-completions.drv' failed to build
error: 1 dependencies of derivation '/nix/store/1zs3847k7mn1n28daz4jbclymwjdmxhi-man-paths.drv' failed to build
error: 1 dependencies of derivation '/nix/store/snv60hjs1p6ni0a4q35bg8m3s5yhq37d-home-manager-generation.drv' failed to build
error: 1 dependencies of derivation '/nix/store/5pq9pmz0rfcq3arrcnmb6jz5vzf7zkzh-user-environment.drv' failed to build
error: 1 dependencies of derivation '/nix/store/iq2bi6q8phrw3bgnvz96jqfs74m17q32-etc.drv' failed to build
error: 1 dependencies of derivation '/nix/store/1yl4nfi9hqlgh8y1yk7s9v3aa02jrlpc-nixos-system-crate-laptop-24.11.20240814.c3aa7b8.drv' failed to build
matt@crate-laptop ~/nix-config (master) [1]> nix log /nix/store/9vxwq4kip2n9gqpm0zshg88xcvqnqx86-hyprpanel.drv
warning: The interpretation of store paths arguments ending in `.drv` recently changed. If this command is now failing try again with '/nix/store/9vxwq4kip2n9gqpm0zshg88xcvqnqx86-hyprpanel.drv^*'
error: build log of '/nix/store/9vxwq4kip2n9gqpm0zshg88xcvqnqx86-hyprpanel.drv' is not available`

**Comment by Jas-SinghFSU** [comment]
@nebunebu @isabelroses Any thoughts?

**Comment by orxngc** [comment]
> Thanks, that got me past those errors! hyprpanel still fails to build; I hope this is helpful:

There is an issue for this > #153

**Pull request merged**